File: soundcloud.js
const cheerio = require("cheerio"),
    axios = require("axios");

function soundcloud(url) {
    return new Promise((async (resolve, reject) => {
        if (!url) reject("Vui lòng nhập url!")
        const res = await axios({
            method: "post",
            url: "https://www.klickaud.co/download.php",
            headers: {
                "Content-Type": "multipart/form-data"
            },
            data: {
                value: url
            }
        }),
            $ = cheerio.load(res.data);
        resolve({
            imageSrc: $('tr[class="mobtable2"]').find("td").find("img").attr("src"),
            songName: $('tr[class="mobtable2"]').find("td:nth-child(2)").text(),
            Quality: $('tr[class="mobtable2"]').find("td:nth-child(3)").text(),
            urlSong: $(".large-centered").find("div[id=dlMP3]").attr("onclick")
                .split("downloadFile")[1]
                .split(",")[0]
                .replace("(", "")
                .replace("'", "")
                .replace("'", "")
        })
    }))
}
soundcloud();

File: soundcloud.py
from tracemalloc import reset_peak
import requests
import json
import pygame
from colorama import Fore
from colorama import Style
url = input("Vui lòng nhập URL cần tải: ")
link = "https://www.klickaud.co/download.php"
res = requests.post(link, data={
    "value": url
})
getUrl = res.text.split('onclick="downloadFile(')[1].split(';')[
    0].split(',')[0].replace("'", "")
# save audio
res1 = requests.get(getUrl)
with open("audio.mp3", "wb") as f:
    f.write(res1.content)
print(Fore.GREEN + "Tải xuống thành công!")
print(Style.RESET_ALL)
reset_peak()
# play audio
pygame.init()
pygame.mixer.init()
pygame.mixer.music.load("audio.mp3")
pygame.mixer.music.play()
while pygame.mixer.music.get_busy():
    pygame.time.Clock().tick(10)
pygame.mixer.music.stop()
pygame.quit()
pygame.mixer.quit()
pygame.display.quit()
pygame.quit()
print(Fore.GREEN + "Xong!")
print(Style.RESET_ALL)
# delete audio
import os
os.remove("audio.mp3")
print(Fore.GREEN + "Xóa thành công!")
print(Style.RESET_ALL)
